lib/src/progress_bar.dart

```diff
@@ -57,74 +57,85 @@ class _VideoProgressBarState extends State<VideoProgressBar> {
-    controller.seekTo(position);
+
+    return position;
+  }
+
+  void _seekToRelativePosition(Offset globalPosition) {
+    controller.seekTo(_calcRelativePosition(context, globalPosition));
   }
 
   @override
   Widget build(BuildContext context) {
-    return GestureDetector(
-      onHorizontalDragStart: (DragStartDetails details) {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        _controllerWasPlaying = controller.value.isPlaying;
-        if (_controllerWasPlaying) {
-          controller.pause();
-        }
-
-        widget.onDragStart?.call();
-      },
-      onHorizontalDragUpdate: (DragUpdateDetails details) {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        // Should only seek if it's not running on Android, or if it is,
-        // then the VideoPlayerController cannot be buffering.
-        // On Android, we need to let the player buffer when scrolling
-        // in order to let the player buffer. https://github.com/flutter/flutter/issues/101409
-        final shouldSeekToRelativePosition =
-            !Platform.isAndroid || !controller.value.isBuffering;
-        if (shouldSeekToRelativePosition) {
-          _seekToRelativePosition(details.globalPosition);
-        }
-
-        widget.onDragUpdate?.call();
-      },
-      onHorizontalDragEnd: (DragEndDetails details) {
-        if (_controllerWasPlaying) {
-          controller.play();
-        }
-
-        widget.onDragEnd?.call();
-      },
-      onTapDown: (TapDownDetails details) {
-        if (!controller.value.isInitialized) {
-          return;
-        }
-        _seekToRelativePosition(details.globalPosition);
-      },
-      child: Center(
-        child: Container(
-          height: MediaQuery.of(context).size.height,
-          width: MediaQuery.of(context).size.width,
-          color: Colors.transparent,
-          child: CustomPaint(
-            painter: _ProgressBarPainter(
-              value: controller.value,
-              colors: widget.colors,
-              barHeight: widget.barHeight,
-              handleHeight: widget.handleHeight,
-              drawShadow: widget.drawShadow,
-            ),
+    final ChewieController chewieController = ChewieController.of(context);
+    final child = Center(
+      child: Container(
+        height: MediaQuery.of(context).size.height,
+        width: MediaQuery.of(context).size.width,
+        color: Colors.transparent,
+        child: CustomPaint(
+          painter: _ProgressBarPainter(
+            value: controller.value,
+            draggableValue: _latestDraggableOffset != null
+                ? _calcRelativePosition(context, _latestDraggableOffset!)
+                : Duration.zero,
+            colors: widget.colors,
+            barHeight: widget.barHeight,
+            handleHeight: widget.handleHeight,
+            drawShadow: widget.drawShadow,
           ),
         ),
       ),
     );
```

Comment:
Again, this should be refactored into a standalone `Widget` class within this file:

```dart
class StaticProgressBar extends StatelessWidget {
  const StaticProgressBar(
    {
      super.key, 
      required this.videoDuration, 
      this.latestDraggableOffset,
    }
  );
  final Duration videoDuration;
  final Offset? latestDraggableOffset;
  
  @override
  Widget build(BuildContext context) {
    return Container(
        height: MediaQuery.of(context).size.height,
        width: MediaQuery.of(context).size.width,
        color: Colors.transparent,
        child: CustomPaint(
          painter: _ProgressBarPainter(
            value: controller.value,
            draggableValue: latestDraggableOffset != null
                ? context.calcRelativePosition(videoDuration, latestDraggableOffset!)
                : Duration.zero,
            colors: widget.colors,
            barHeight: widget.barHeight,
            handleHeight: widget.handleHeight,
            drawShadow: widget.drawShadow,
          ),
        ),
      );
  }  
}
```

Then you would just create an instance of that class as your child:

```dart
final child = Center(
  child: StaticProgressBar(
    videoDuration: controller.value.duration, 
    latestDraggableOffset: _latestDraggableOffset,
  ),
);
```

Doing it this way makes it easy to test this component in isolation in the long term.